docs/source/start/declarative.rst

```diff
@@ -184,6 +187,44 @@ e.g. numeric properties. This example changes the ``min_card`` property of an
        min_card: 0
        max_card: .inf
 
+Extending container attributes of objects
+-----------------------------------------
```

Comment:
I didn't know that this behavior exists and I find it just a bit missleading/unintuitive. For me `create` means that a new object is created... but one could also argue that used in combination with !uuid/!promise only a new reference is created. To complete the logic of the latter, `delete` should also distinguish between only deleting the reference (href from attribute, or w/e) and really deleting it. That means wiping the model from the objects existence without leaving broken references behind. This would happen if the user calls `delete` on an attribute where the `ProxyAccessor`(parent/child relationship) is used.

So either we keep this behavior of `create`, add a subsection to inform about this feature in the docs and make sure on `delete` completeness described above. Or we take the verbose route and add `extend` (maybe better: `create-reference`/`delete-reference`)... where I'd need to also implement the `remove` (`delete-reference`) operation.

Reply:
> I didn't know that this behavior exists and I find it just a bit misleading/unintuitive.

That's what I meant with my last paragraph - not quite what you'd call the "obvious way to do it". :) I agree we should at least mention that it's a thing in the docs somewhere.

> make sure on `delete` completeness

This part should rather go into the base library, as it's not really specific to decl's `delete` operation. If you delete something with just `del my_func.functions[0]`, you'd probably expect that no broken refs stay behind (which is also Capella's behavior, by the way). So, yes, we should implement it, but not here.

This might be a somewhat bigger undertaking though, because we'd need to make an index of who links where if we don't want to completely destroy runtime performance, and whoever deletes also needs to understand how much to delete (e.g. the LinkAccessor needs to delete the entire XML element, while the AttrProxyAccessor only removes one UUID from one attribute).

> Or we take the verbose route and add `extend` (maybe better: `create-reference`/`delete-reference`)... where I'd need to also implement the `remove` (`delete-reference`) operation.

Not entirely sure what you're aiming at. Do you want to split up "create" into "create-object" and "create-reference", and then add "extend" on top for moving stuff around? Not a big fan of that, it just makes things unnecessarily complex for the end user *and* the implementation.

Currently there's no easy and reliable way to distinguish between the cases anyway. Objects get resolved successively as they become available, and by the time we finally get to act on one, we don't know anymore whether the original YAML said `!uuid`, `!promise`, or had a dictionary. I'd say this approach isn't feasible without rewriting large parts of the logic.